# Profitable Application Profiles for IOS and Android App Stores

In this project I am a data analyst working for a company that specializes in mobile app development. My company strictly deploys free-to-play (F2P) strategy games. 

Since our games are F2P, we need to earn revenue somehow. Our main source of this revenue is through in-app ads. Hence, the more users that see and engage with our ads, the higher our revenue. I have been tasked with analyzing whether or not we could generate more installs if we branched out into different game genres.

In [1]:
import pandas as pd

apple_df = pd.read_csv('AppleStore.csv')
google_df = pd.read_csv('googleplaystore.csv')

Above, we opened up two different datasets that we will explore below.

We are going to start doing some initial exploration of our dataset by inspecting the column headers and a few rows of each dataset. We will dive into the apple dataset first.

In [2]:
apple_df

id                                    track_name  size_bytes  \
0      284882215                                      Facebook   389879808   
1      389801252                                     Instagram   113954816   
2      529479190                                Clash of Clans   116476928   
3      420009108                                    Temple Run    65921024   
4      284035177                       Pandora - Music & Radio   130242560   
...          ...                                           ...         ...   
7192  1170406182  Shark Boom - Challenge Friends with your Pet   245415936   
7193  1069830936                           【謎解き】ヤミすぎ彼女からのメッセージ    16808960   
7194  1070052833                                    Go!Go!Cat!    91468800   
7195  1081295232   Suppin Detective: Expose their true visage!    83026944   
7196   977965019           みんなのお弁当 by クックパッド ~お弁当をレシピ付きで記録・共有~    51174400   

     currency  price  rating_count_tot  rating_count_ver  user_rating  \
0         USD    0.0           2974676               212          3.5   
1         USD    0.0           2161558              1289          4.5   
2         USD    0.0           2130805               579          4.5   
3         USD    0.0           1724546              3842          4.5   
4         USD    0.0           1126879              3594          4.0   
...       ...    ...               ...               ...          ...   
7192      USD    0.0                 0                 0          0.0   
7193      USD    0.0                 0                 0          0.0   
7194      USD    0.0                 0                 0          0.0   
7195      USD    0.0                 0                 0          0.0   
7196      USD    0.0                 0                 0          0.0   

      user_rating_ver      ver cont_rating        prime_genre  \
0                 3.5     95.0          4+  Social Networking   
1                 4.0    10.23         12+      Photo & Video   
2                 4.5  9.24.12          9+              Games   
3                 4.0    1.6.2          9+              Games   
4                 4.5    8.4.1         12+              Music   
...               ...      ...         ...                ...   
7192              0.0    1.0.9          4+              Games   
7193              0.0      1.2          9+               Book   
7194              0.0    1.1.2         12+              Games   
7195              0.0    1.0.3         12+      Entertainment   
7196              0.0    1.4.0          4+       Food & Drink   

      sup_devices.num  ipadSc_urls.num  lang.num  vpp_lic  
0                  37                1        29        1  
1                  37                0        29        1  
2                  38                5        18        1  
3                  40                5         1        1  
4                  37                4         1        1  
...               ...              ...       ...      ...  
7192               38                5         1        1  
7193               38                0         1        1  
7194               37                2         2        1  
7195               40                0         1        1  
7196               37                0         1        1  

[7197 rows x 16 columns]

This also reveals that there are 7197 rows and 16 columns.
Now, we'll do the same the google play store dataset

In [3]:
google_df

App             Category  \
0         Photo Editor & Candy Camera & Grid & ScrapBook       ART_AND_DESIGN   
1                                    Coloring book moana       ART_AND_DESIGN   
2      U Launcher Lite – FREE Live Cool Themes, Hide ...       ART_AND_DESIGN   
3                                  Sketch - Draw & Paint       ART_AND_DESIGN   
4                  Pixel Draw - Number Art Coloring Book       ART_AND_DESIGN   
...                                                  ...                  ...   
10836                                   Sya9a Maroc - FR               FAMILY   
10837                   Fr. Mike Schmitz Audio Teachings               FAMILY   
10838                             Parkinson Exercices FR              MEDICAL   
10839                      The SCP Foundation DB fr nn5n  BOOKS_AND_REFERENCE   
10840      iHoroscope - 2018 Daily Horoscope & Astrology            LIFESTYLE   

       Rating Reviews                Size     Installs  Type Price  \
0         4.1     159                 19M      10,000+  Free     0   
1         3.9     967                 14M     500,000+  Free     0   
2         4.7   87510                8.7M   5,000,000+  Free     0   
3         4.5  215644                 25M  50,000,000+  Free     0   
4         4.3     967                2.8M     100,000+  Free     0   
...       ...     ...                 ...          ...   ...   ...   
10836     4.5      38                 53M       5,000+  Free     0   
10837     5.0       4                3.6M         100+  Free     0   
10838     NaN       3                9.5M       1,000+  Free     0   
10839     4.5     114  Varies with device       1,000+  Free     0   
10840     4.5  398307                 19M  10,000,000+  Free     0   

      Content Rating                     Genres      Last Updated  \
0           Everyone               Art & Design   January 7, 2018   
1           Everyone  Art & Design;Pretend Play  January 15, 2018   
2           Everyone               Art & Design    August 1, 2018   
3               Teen               Art & Design      June 8, 2018   
4           Everyone    Art & Design;Creativity     June 20, 2018   
...              ...                        ...               ...   
10836       Everyone                  Education     July 25, 2017   
10837       Everyone                  Education      July 6, 2018   
10838       Everyone                    Medical  January 20, 2017   
10839     Mature 17+          Books & Reference  January 19, 2015   
10840       Everyone                  Lifestyle     July 25, 2018   

              Current Ver         Android Ver  
0                   1.0.0        4.0.3 and up  
1                   2.0.0        4.0.3 and up  
2                   1.2.4        4.0.3 and up  
3      Varies with device          4.2 and up  
4                     1.1          4.4 and up  
...                   ...                 ...  
10836                1.48          4.1 and up  
10837                 1.0          4.1 and up  
10838                 1.0          2.2 and up  
10839  Varies with device  Varies with device  
10840  Varies with device  Varies with device  

[10841 rows x 13 columns]

Our datasets vary in their structures. In our apple dataset, the following columns sound potentially useful in achieving our analysis: ['id', 'track_name', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user,'prime_genre']. This was determined through a combination of some columns being intuitive, and understanding others through the [datasets documentation](https://www.kaggle.com/datasets/ramamet4/app-store-apple-data-set-10k-apps) available on Kaggle.

Switching over to the [google play store dataset](https://www.kaggle.com/datasets/lava18/google-play-store-apps), the documentation is not as informative, but thankfully the columns are mostly intuitive. I belive the following columns may be useful: ['App', 'Category', 'Reviews', 'Installs', 'Type', 'Genre', 'Current Ver']. One thing I noticed is that this dataset is a little less analysis-friendly and will need some extra cleaning. We will get working on that after a while.

Another useful initial exploratory function to use is the DataFrame.describe() method, as seen below:

In [4]:
apple_df.describe()

id    size_bytes        price  rating_count_tot  \
count  7.197000e+03  7.197000e+03  7197.000000      7.197000e+03   
mean   8.631310e+08  1.991345e+08     1.726218      1.289291e+04   
std    2.712368e+08  3.592069e+08     5.833006      7.573941e+04   
min    2.816565e+08  5.898240e+05     0.000000      0.000000e+00   
25%    6.000937e+08  4.692275e+07     0.000000      2.800000e+01   
50%    9.781482e+08  9.715302e+07     0.000000      3.000000e+02   
75%    1.082310e+09  1.819249e+08     1.990000      2.793000e+03   
max    1.188376e+09  4.025970e+09   299.990000      2.974676e+06   

       rating_count_ver  user_rating  user_rating_ver  sup_devices.num  \
count       7197.000000  7197.000000      7197.000000      7197.000000   
mean         460.373906     3.526956         3.253578        37.361817   
std         3920.455183     1.517948         1.809363         3.737715   
min            0.000000     0.000000         0.000000         9.000000   
25%            1.000000     3.500000         2.500000        37.000000   
50%           23.000000     4.000000         4.000000        37.000000   
75%          140.000000     4.500000         4.500000        38.000000   
max       177050.000000     5.000000         5.000000        47.000000   

       ipadSc_urls.num     lang.num      vpp_lic  
count      7197.000000  7197.000000  7197.000000  
mean          3.707100     5.434903     0.993053  
std           1.986005     7.919593     0.083066  
min           0.000000     0.000000     0.000000  
25%           3.000000     1.000000     1.000000  
50%           5.000000     1.000000     1.000000  
75%           5.000000     8.000000     1.000000  
max           5.000000    75.000000     1.000000

In [5]:
google_df.describe()

Rating
count  9367.000000
mean      4.193338
std       0.537431
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max      19.000000

Notice how the output for the google dataset contains data on much fewer columns. This is because many of the columns have odd formatting such as representing size as 14M instead of 14000000, and adding '+' symbols after the number of installs.

# Data Cleaning

# Invalid and duplicate data

Scanning over the above output, there is something that immedietly seems out of place. The rating column has a max value of 19, while the other statistics and intuition indicates that it should max at 5. After validating something is off, we will remove this row from the data.

In [6]:
google_df[google_df['Rating'] == 19]

App Category  Rating Reviews  \
10472  Life Made WI-Fi Touchscreen Photo Frame      1.9    19.0    3.0M   

         Size Installs Type     Price Content Rating             Genres  \
10472  1,000+     Free    0  Everyone            NaN  February 11, 2018   

      Last Updated Current Ver Android Ver  
10472       1.0.19  4.0 and up         NaN

In [7]:
google_df.drop(10472, inplace=True)

In [8]:
google_df.describe()

Rating
count  9366.000000
mean      4.191757
std       0.515219
min       1.000000
25%       4.000000
50%       4.300000
75%       4.500000
max       5.000000

Deleting that one row makes the max value drop down to what we would expect.

One crucial step in the data cleaning process is to check for duplicate rows in a dataframe. It could be possible that we have data on one app for different versions, so initially I will check for duplicate app names, versus entire rows that would not be detected as duplicates for different versions.

In [9]:
apple_df[apple_df.duplicated('track_name')]

id           track_name  size_bytes currency  price  \
4463  1178454060  Mannequin Challenge    59572224      USD    0.0   
4831  1089824278    VR Roller Coaster   240964608      USD    0.0   

      rating_count_tot  rating_count_ver  user_rating  user_rating_ver    ver  \
4463               105                58          4.0              4.5  1.0.1   
4831                67                44          3.5              4.0   0.81   

     cont_rating prime_genre  sup_devices.num  ipadSc_urls.num  lang.num  \
4463          4+       Games               38                5         1   
4831          4+       Games               38                0         1   

      vpp_lic  
4463        1  
4831        1

In [10]:
google_df[google_df.duplicated('App')]

App       Category  Rating Reviews  \
229          Quick PDF Scanner + OCR FREE       BUSINESS     4.2   80805   
236                                   Box       BUSINESS     4.2  159872   
239                    Google My Business       BUSINESS     4.4   70991   
256                   ZOOM Cloud Meetings       BUSINESS     4.4   31614   
261             join.me - Simple Meetings       BUSINESS     4.0    6989   
...                                   ...            ...     ...     ...   
10715                  FarmersOnly Dating         DATING     3.0    1145   
10720  Firefox Focus: The privacy browser  COMMUNICATION     4.4   36981   
10730                         FP Notebook        MEDICAL     4.5     410   
10753      Slickdeals: Coupons & Shopping       SHOPPING     4.5   33599   
10768                                AAFP        MEDICAL     3.8      63   

                     Size     Installs  Type Price Content Rating  \
229    Varies with device   5,000,000+  Free     0       Everyone   
236    Varies with device  10,000,000+  Free     0       Everyone   
239    Varies with device   5,000,000+  Free     0       Everyone   
256                   37M  10,000,000+  Free     0       Everyone   
261    Varies with device   1,000,000+  Free     0       Everyone   
...                   ...          ...   ...   ...            ...   
10715                1.4M     100,000+  Free     0     Mature 17+   
10720                4.0M   1,000,000+  Free     0       Everyone   
10730                 60M      50,000+  Free     0       Everyone   
10753                 12M   1,000,000+  Free     0       Everyone   
10768                 24M      10,000+  Free     0       Everyone   

              Genres       Last Updated         Current Ver  \
229         Business  February 26, 2018  Varies with device   
236         Business      July 31, 2018  Varies with device   
239         Business      July 24, 2018    2.19.0.204537701   
256         Business      July 20, 2018      4.1.28165.0716   
261         Business      July 16, 2018           4.3.0.508   
...              ...                ...                 ...   
10715         Dating  February 25, 2016                 2.2   
10720  Communication       July 6, 2018                 5.2   
10730        Medical     March 24, 2018           2.1.0.372   
10753       Shopping      July 30, 2018                 3.9   
10768        Medical      June 22, 2018               2.3.1   

              Android Ver  
229          4.0.3 and up  
236    Varies with device  
239            4.4 and up  
256            4.0 and up  
261            4.4 and up  
...                   ...  
10715          4.0 and up  
10720          5.0 and up  
10730          4.4 and up  
10753          4.4 and up  
10768          5.0 and up  

[1181 rows x 13 columns]

This shows that our apple dataset contains 2 duplicated rows, and the google one contains 1181 duplicates. Note that this means the dataset contains 1181 apps that appear more than once, versus duplicate rows accounting for 1181 terms. In other terms, if an app name appears twice or twelve times, it is only counted as a duplicate once in our total number of duplicate entries.

In [11]:
display(apple_df[apple_df['track_name'] == 'Mannequin Challenge'])
display(google_df[google_df['App'] == 'Box'])

id           track_name  size_bytes currency  price  \
2948  1173990889  Mannequin Challenge   109705216      USD    0.0   
4463  1178454060  Mannequin Challenge    59572224      USD    0.0   

      rating_count_tot  rating_count_ver  user_rating  user_rating_ver    ver  \
2948               668                87          3.0              3.0    1.4   
4463               105                58          4.0              4.5  1.0.1   

     cont_rating prime_genre  sup_devices.num  ipadSc_urls.num  lang.num  \
2948          9+       Games               37                4         1   
4463          4+       Games               38                5         1   

      vpp_lic  
2948        1  
4463        1

App  Category  Rating Reviews                Size     Installs  Type  \
204  Box  BUSINESS     4.2  159872  Varies with device  10,000,000+  Free   
236  Box  BUSINESS     4.2  159872  Varies with device  10,000,000+  Free   
265  Box  BUSINESS     4.2  159872  Varies with device  10,000,000+  Free   

    Price Content Rating    Genres   Last Updated         Current Ver  \
204     0       Everyone  Business  July 31, 2018  Varies with device   
236     0       Everyone  Business  July 31, 2018  Varies with device   
265     0       Everyone  Business  July 31, 2018  Varies with device   

            Android Ver  
204  Varies with device  
236  Varies with device  
265  Varies with device

I would have liked to see each value have a unique 'last updated' or version value, but that does not appear to be the case. It appears data has been gathered for the apps at different times, but not necessarily different verisons. 
I will instead take the number of reviews for each app into consideration when removing duplicates. I believe this is the best available approach for two reasons:

    1. The count of reviews is readily available in both datasets.
    
    2. Choosing the data with the highest reviews offers us more reliable ratings than if we had a 
       smaller sample.

The following code drops duplicates, while keeping the row with the highest number of reviews. In the case that a duplicate contains the same number of reviews, the first row is kept.

In [12]:
apple_df.sort_values('rating_count_tot', ascending=False, inplace=True) #Sort in non-increasing order
apple_df.drop_duplicates('track_name', keep='first', inplace=True)

google_df.sort_values('Reviews', ascending=False, inplace=True) #Sort in non-increasing order
google_df.drop_duplicates('App', keep='first', inplace=True)

display(apple_df[apple_df.duplicated('track_name')])
display(google_df[google_df.duplicated('App')])

Empty DataFrame
Columns: [id, track_name, size_bytes, currency, price, rating_count_tot, rating_count_ver, user_rating, user_rating_ver, ver, cont_rating, prime_genre, sup_devices.num, ipadSc_urls.num, lang.num, vpp_lic]
Index: []

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

As seen by the above display statements, we have no more duplicates.

In [13]:
google_df

App          Category  Rating Reviews  \
2989             GollerCepte Live Score            SPORTS     4.2    9992   
4970       Ad Block REMOVER - NEED ROOT             TOOLS     3.3     999   
2723                SnipSnap Coupon App          SHOPPING     4.2    9975   
3079  US Open Tennis Championships 2018            SPORTS     4.0    9971   
3229                         DreamTrips  TRAVEL_AND_LOCAL     4.7    9971   
...                                 ...               ...     ...     ...   
2465                  ear super hearing           MEDICAL     NaN       0   
9925                      Reisedealz.eu            SOCIAL     NaN       0   
9928                        EU Whoiswho          BUSINESS     NaN       0   
9929     EU Brazil Green Business Forum      PRODUCTIVITY     NaN       0   
7101                 Register.ca Mobile      PRODUCTIVITY     NaN       0   

      Size    Installs  Type Price Content Rating          Genres  \
2989   31M  1,000,000+  Free     0       Everyone          Sports   
4970   91k    100,000+  Free     0       Everyone           Tools   
2723   18M  1,000,000+  Free     0       Everyone        Shopping   
3079   33M  1,000,000+  Free     0       Everyone          Sports   
3229   22M    500,000+  Free     0           Teen  Travel & Local   
...    ...         ...   ...   ...            ...             ...   
2465  1.4M          5+  Free     0       Everyone         Medical   
9925   10M         10+  Free     0       Everyone          Social   
9928  2.7M         10+  Free     0       Everyone        Business   
9929  8.7M         10+  Free     0       Everyone    Productivity   
7101  1.2M        100+  Free     0       Everyone    Productivity   

            Last Updated            Current Ver Android Ver  
2989        May 23, 2018                    6.5  4.1 and up  
4970   December 17, 2013                    3.2  2.2 and up  
2723    January 22, 2018                    1.4  4.3 and up  
3079        June 5, 2018                    7.1  5.0 and up  
3229      August 6, 2018                 1.28.1  5.0 and up  
...                  ...                    ...         ...  
2465       July 31, 2018  1.0 Super Ear Hearing  4.0 and up  
9925  September 24, 2015                    1.0  4.0 and up  
9928    December 1, 2016                  0.0.1  4.0 and up  
9929      April 18, 2017                  1.7.1  2.1 and up  
7101     August 19, 2016                    2.0  5.0 and up  

[9659 rows x 13 columns]

# Irrelevant data

### Non-English Apps
My company is based in the United States, and we don't branch out into other nations. Due to this, we do not care about apps based in foreign countries. We will remove rows with apps in languages other than english. 

A really neat approach to removing non-english apps is to remove apps that dont fully fall into the ascii alphabet. Unfortunately, some apps might contain valid, non-ascii characters such as trademarks. To avoid removing unecessary data, we will only remove rows containing 4 or more non-ascii characters. This should work almost flawlessly, as most apps contain more than 3 characters in their name.

In [14]:
def count_non_ascii_chars(text):
    non_ascii_count = sum(1 for c in text if not c.isascii())
    return non_ascii_count

In [15]:
display(apple_df)
apple_df = apple_df[apple_df['track_name'].map(lambda x: count_non_ascii_chars(x) <= 3)].copy()
display(apple_df)

id                           track_name  size_bytes currency  \
0      284882215                             Facebook   389879808      USD   
1      389801252                            Instagram   113954816      USD   
2      529479190                       Clash of Clans   116476928      USD   
3      420009108                           Temple Run    65921024      USD   
4      284035177              Pandora - Music & Radio   130242560      USD   
...          ...                                  ...         ...      ...   
6581  1040105227                          全国一斉性格テストDX    18067456      USD   
6582  1040366566                            盗墓之王-寻龙秘事   208276480      USD   
6583  1040417757         Bibi & Tina: Pferdeabenteuer   622986240      USD   
6584  1097148221                              S ou SS     4824064      USD   
7196   977965019  みんなのお弁当 by クックパッド ~お弁当をレシピ付きで記録・共有~    51174400      USD   

      price  rating_count_tot  rating_count_ver  user_rating  user_rating_ver  \
0      0.00           2974676               212          3.5              3.5   
1      0.00           2161558              1289          4.5              4.0   
2      0.00           2130805               579          4.5              4.5   
3      0.00           1724546              3842          4.5              4.0   
4      0.00           1126879              3594          4.0              4.5   
...     ...               ...               ...          ...              ...   
6581   0.00                 0                 0          0.0              0.0   
6582   0.00                 0                 0          0.0              0.0   
6583   2.99                 0                 0          0.0              0.0   
6584   2.99                 0                 0          0.0              0.0   
7196   0.00                 0                 0          0.0              0.0   

          ver cont_rating        prime_genre  sup_devices.num  \
0        95.0          4+  Social Networking               37   
1       10.23         12+      Photo & Video               37   
2     9.24.12          9+              Games               38   
3       1.6.2          9+              Games               40   
4       8.4.1         12+              Music               37   
...       ...         ...                ...              ...   
6581      1.6         17+      Entertainment               40   
6582   1.0.29         12+              Games               37   
6583    1.2.1          4+              Games               38   
6584    1.0.0          4+          Education               38   
7196    1.4.0          4+       Food & Drink               37   

      ipadSc_urls.num  lang.num  vpp_lic  
0                   1        29        1  
1                   0        29        1  
2                   5        18        1  
3                   5         1        1  
4                   4         1        1  
...               ...       ...      ...  
6581                0         1        1  
6582                5         2        1  
6583                5         1        1  
6584                5         1        1  
7196                0         1        1  

[7195 rows x 16 columns]

id                    track_name  size_bytes currency  price  \
0      284882215                      Facebook   389879808      USD   0.00   
1      389801252                     Instagram   113954816      USD   0.00   
2      529479190                Clash of Clans   116476928      USD   0.00   
3      420009108                    Temple Run    65921024      USD   0.00   
4      284035177       Pandora - Music & Radio   130242560      USD   0.00   
...          ...                           ...         ...      ...    ...   
6566   897880328                         EOPAN    34338816      USD   0.00   
6572   895700579                    大富豪 Online   187623424      USD   0.00   
6575  1132440751                      Wi-Fiミレル    30086144      USD   0.00   
6583  1040417757  Bibi & Tina: Pferdeabenteuer   622986240      USD   2.99   
6584  1097148221                       S ou SS     4824064      USD   2.99   

      rating_count_tot  rating_count_ver  user_rating  user_rating_ver  \
0              2974676               212          3.5              3.5   
1              2161558              1289          4.5              4.0   
2              2130805               579          4.5              4.5   
3              1724546              3842          4.5              4.0   
4              1126879              3594          4.0              4.5   
...                ...               ...          ...              ...   
6566                 0                 0          0.0              0.0   
6572                 0                 0          0.0              0.0   
6575                 0                 0          0.0              0.0   
6583                 0                 0          0.0              0.0   
6584                 0                 0          0.0              0.0   

          ver cont_rating        prime_genre  sup_devices.num  \
0        95.0          4+  Social Networking               37   
1       10.23         12+      Photo & Video               37   
2     9.24.12          9+              Games               38   
3       1.6.2          9+              Games               40   
4       8.4.1         12+              Music               37   
...       ...         ...                ...              ...   
6566    2.2.0          4+      Photo & Video               37   
6572  1.2.299          4+              Games               40   
6575    1.1.1          4+          Utilities               37   
6583    1.2.1          4+              Games               38   
6584    1.0.0          4+          Education               38   

      ipadSc_urls.num  lang.num  vpp_lic  
0                   1        29        1  
1                   0        29        1  
2                   5        18        1  
3                   5         1        1  
4                   4         1        1  
...               ...       ...      ...  
6566                1         2        1  
6572                5         1        1  
6575                3         1        0  
6583                5         1        1  
6584                5         1        1  

[6181 rows x 16 columns]

In [16]:
display(google_df)
google_df = google_df[google_df['App'].map(lambda x: count_non_ascii_chars(x) <= 3)].copy()
display(google_df)

App          Category  Rating Reviews  \
2989             GollerCepte Live Score            SPORTS     4.2    9992   
4970       Ad Block REMOVER - NEED ROOT             TOOLS     3.3     999   
2723                SnipSnap Coupon App          SHOPPING     4.2    9975   
3079  US Open Tennis Championships 2018            SPORTS     4.0    9971   
3229                         DreamTrips  TRAVEL_AND_LOCAL     4.7    9971   
...                                 ...               ...     ...     ...   
2465                  ear super hearing           MEDICAL     NaN       0   
9925                      Reisedealz.eu            SOCIAL     NaN       0   
9928                        EU Whoiswho          BUSINESS     NaN       0   
9929     EU Brazil Green Business Forum      PRODUCTIVITY     NaN       0   
7101                 Register.ca Mobile      PRODUCTIVITY     NaN       0   

      Size    Installs  Type Price Content Rating          Genres  \
2989   31M  1,000,000+  Free     0       Everyone          Sports   
4970   91k    100,000+  Free     0       Everyone           Tools   
2723   18M  1,000,000+  Free     0       Everyone        Shopping   
3079   33M  1,000,000+  Free     0       Everyone          Sports   
3229   22M    500,000+  Free     0           Teen  Travel & Local   
...    ...         ...   ...   ...            ...             ...   
2465  1.4M          5+  Free     0       Everyone         Medical   
9925   10M         10+  Free     0       Everyone          Social   
9928  2.7M         10+  Free     0       Everyone        Business   
9929  8.7M         10+  Free     0       Everyone    Productivity   
7101  1.2M        100+  Free     0       Everyone    Productivity   

            Last Updated            Current Ver Android Ver  
2989        May 23, 2018                    6.5  4.1 and up  
4970   December 17, 2013                    3.2  2.2 and up  
2723    January 22, 2018                    1.4  4.3 and up  
3079        June 5, 2018                    7.1  5.0 and up  
3229      August 6, 2018                 1.28.1  5.0 and up  
...                  ...                    ...         ...  
2465       July 31, 2018  1.0 Super Ear Hearing  4.0 and up  
9925  September 24, 2015                    1.0  4.0 and up  
9928    December 1, 2016                  0.0.1  4.0 and up  
9929      April 18, 2017                  1.7.1  2.1 and up  
7101     August 19, 2016                    2.0  5.0 and up  

[9659 rows x 13 columns]

App          Category  Rating Reviews  \
2989             GollerCepte Live Score            SPORTS     4.2    9992   
4970       Ad Block REMOVER - NEED ROOT             TOOLS     3.3     999   
2723                SnipSnap Coupon App          SHOPPING     4.2    9975   
3079  US Open Tennis Championships 2018            SPORTS     4.0    9971   
3229                         DreamTrips  TRAVEL_AND_LOCAL     4.7    9971   
...                                 ...               ...     ...     ...   
2465                  ear super hearing           MEDICAL     NaN       0   
9925                      Reisedealz.eu            SOCIAL     NaN       0   
9928                        EU Whoiswho          BUSINESS     NaN       0   
9929     EU Brazil Green Business Forum      PRODUCTIVITY     NaN       0   
7101                 Register.ca Mobile      PRODUCTIVITY     NaN       0   

      Size    Installs  Type Price Content Rating          Genres  \
2989   31M  1,000,000+  Free     0       Everyone          Sports   
4970   91k    100,000+  Free     0       Everyone           Tools   
2723   18M  1,000,000+  Free     0       Everyone        Shopping   
3079   33M  1,000,000+  Free     0       Everyone          Sports   
3229   22M    500,000+  Free     0           Teen  Travel & Local   
...    ...         ...   ...   ...            ...             ...   
2465  1.4M          5+  Free     0       Everyone         Medical   
9925   10M         10+  Free     0       Everyone          Social   
9928  2.7M         10+  Free     0       Everyone        Business   
9929  8.7M         10+  Free     0       Everyone    Productivity   
7101  1.2M        100+  Free     0       Everyone    Productivity   

            Last Updated            Current Ver Android Ver  
2989        May 23, 2018                    6.5  4.1 and up  
4970   December 17, 2013                    3.2  2.2 and up  
2723    January 22, 2018                    1.4  4.3 and up  
3079        June 5, 2018                    7.1  5.0 and up  
3229      August 6, 2018                 1.28.1  5.0 and up  
...                  ...                    ...         ...  
2465       July 31, 2018  1.0 Super Ear Hearing  4.0 and up  
9925  September 24, 2015                    1.0  4.0 and up  
9928    December 1, 2016                  0.0.1  4.0 and up  
9929      April 18, 2017                  1.7.1  2.1 and up  
7101     August 19, 2016                    2.0  5.0 and up  

[9614 rows x 13 columns]

As seen above, removing non english apps reduced our apple dataset from 7195 to 6181 rows. Our google dataset was only reduced by about 45 rows.

# Excluding apps from irrelevant genres
As mentioned earlier, we create only games, so will remove applications that don't fit this requirement.

In [17]:
apple_df = apple_df.loc[apple_df['prime_genre'] == 'Games']

There are two columns in the google dataset that look like they could be what we want to filter on, Genres and Category

In [18]:
display(google_df['Genres'].value_counts())
display(google_df['Category'].value_counts())

Tools                                  826
Entertainment                          557
Education                              503
Business                               419
Medical                                395
                                      ... 
Tools;Education                          1
Entertainment;Education                  1
Parenting;Brain Games                    1
Video Players & Editors;Creativity       1
Health & Fitness;Action & Adventure      1
Name: Genres, Length: 119, dtype: int64

FAMILY                 1861
GAME                    941
TOOLS                   827
BUSINESS                419
MEDICAL                 395
PERSONALIZATION         375
PRODUCTIVITY            373
LIFESTYLE               364
FINANCE                 345
SPORTS                  325
COMMUNICATION           314
HEALTH_AND_FITNESS      288
PHOTOGRAPHY             280
NEWS_AND_MAGAZINES      250
SOCIAL                  239
TRAVEL_AND_LOCAL        219
BOOKS_AND_REFERENCE     218
SHOPPING                201
DATING                  171
VIDEO_PLAYERS           163
MAPS_AND_NAVIGATION     129
FOOD_AND_DRINK          112
EDUCATION               107
ENTERTAINMENT            86
LIBRARIES_AND_DEMO       84
AUTO_AND_VEHICLES        84
WEATHER                  79
HOUSE_AND_HOME           73
EVENTS                   64
ART_AND_DESIGN           60
PARENTING                60
COMICS                   55
BEAUTY                   53
Name: Category, dtype: int64

The difference between the two is not necessarily intuitive, but it appears that the Genres column is much more granular. We are looking into other types of games, so we want to look at all games and will use 'Category'. However, it seems odd to me that 'GAME' only accounts for about 9% of the apps on the google store. After some quick researching, it appears that the FAMILY category is used for calssifying games for younger children, so we will want to include that.

In [19]:
google_df.loc[(google_df['Category'] == 'FAMILY') & (google_df['Rating'] >=4.5)]

App Category  Rating  \
5326                                      Al jazeera TV   FAMILY     4.6   
6996               Art Pixel Coloring. Color by Number.   FAMILY     4.5   
4840                              Z-Empire: Dead Strike   FAMILY     4.6   
2145                   Sworkit Kids - Fitness Meets Fun   FAMILY     4.7   
2025                             Princess Coloring Book   FAMILY     4.5   
...                                                 ...      ...     ...   
8018  30WPM Amateur ham radio Koch CW Morse code tra...   FAMILY     5.0   
7204          TI-84 CE Graphing Calculator Manual TI 84   FAMILY     5.0   
7139                                        CB Register   FAMILY     5.0   
5246                              Hey AJ! It's Bedtime!   FAMILY     5.0   
5268                            AJ Rafael Music Lessons   FAMILY     5.0   

     Reviews  Size    Installs  Type  Price Content Rating  \
5326    9952  3.7M    100,000+  Free      0           Teen   
6996     992   49M    100,000+  Free      0           Teen   
4840      99   19M     10,000+  Free      0           Teen   
2145    9856   55M    100,000+  Free      0       Everyone   
2025    9779   39M  5,000,000+  Free      0       Everyone   
...      ...   ...         ...   ...    ...            ...   
8018       1  3.7M         10+  Paid  $1.49       Everyone   
7204       1   27M        100+  Paid  $4.99       Everyone   
7139       1  5.6M         10+  Free      0       Everyone   
5246       1   63M         10+  Paid  $4.99       Everyone   
5268       1  8.2M         10+  Free      0       Everyone   

                          Genres       Last Updated Current Ver   Android Ver  
5326               Entertainment       July 5, 2016       1.0.0  4.0.3 and up  
6996               Entertainment     August 2, 2018   1.2.0.200    4.2 and up  
4840                    Strategy       June 8, 2018       1.8.7  4.0.3 and up  
2145  Health & Fitness;Education   February 2, 2016       1.1.1    4.0 and up  
2025        Education;Creativity  February 25, 2018       1.2.8    4.0 and up  
...                          ...                ...         ...           ...  
8018                   Education       May 18, 2018       2.0.2    2.1 and up  
7204                   Education     March 28, 2018       1.5.2    4.1 and up  
7139               Entertainment      June 27, 2018         1.0    4.1 and up  
5246                   Education      April 4, 2018      1.0.59    2.3 and up  
5268               Entertainment  November 29, 2017       1.2.1    4.1 and up  

[510 rows x 13 columns]

In [20]:
google_df = google_df.loc[(google_df['Category'] == 'GAME') | (google_df['Category'] == 'FAMILY')]
display(apple_df)
display(google_df)

id                    track_name  size_bytes currency  price  \
2      529479190                Clash of Clans   116476928      USD   0.00   
3      420009108                    Temple Run    65921024      USD   0.00   
7      553834731              Candy Crush Saga   222846976      USD   0.00   
9      343200656                   Angry Birds   175966208      USD   0.00   
10     512939461                Subway Surfers   156038144      USD   0.00   
...          ...                           ...         ...      ...    ...   
6602  1043450394  BrainTraining!!GASHAAAAAAN!!   132264960      USD   0.00   
6585   967870967                 Cyber Reversi    52166656      USD   0.00   
6560  1038379658        Green Game TimeSwapper   263876608      USD   2.99   
6572   895700579                    大富豪 Online   187623424      USD   0.00   
6583  1040417757  Bibi & Tina: Pferdeabenteuer   622986240      USD   2.99   

      rating_count_tot  rating_count_ver  user_rating  user_rating_ver  \
2              2130805               579          4.5              4.5   
3              1724546              3842          4.5              4.0   
7               961794              2453          4.5              4.5   
9               824451               107          4.5              3.0   
10              706110                97          4.5              4.0   
...                ...               ...          ...              ...   
6602                 0                 0          0.0              0.0   
6585                 0                 0          0.0              0.0   
6560                 0                 0          0.0              0.0   
6572                 0                 0          0.0              0.0   
6583                 0                 0          0.0              0.0   

          ver cont_rating prime_genre  sup_devices.num  ipadSc_urls.num  \
2     9.24.12          9+       Games               38                5   
3       1.6.2          9+       Games               40                5   
7     1.101.0          4+       Games               43                5   
9       7.4.0          4+       Games               38                0   
10     1.72.1          9+       Games               38                5   
...       ...         ...         ...              ...              ...   
6602    1.9.2          4+       Games               38                2   
6585    1.0.4          4+       Games               40                5   
6560    1.0.0          4+       Games               40                5   
6572  1.2.299          4+       Games               40                5   
6583    1.2.1          4+       Games               38                5   

      lang.num  vpp_lic  
2           18        1  
3            1        1  
7           24        1  
9           10        1  
10           1        1  
...        ...      ...  
6602         1        1  
6585         4        1  
6560         1        1  
6572         1        1  
6583         1        1  

[3390 rows x 16 columns]

App Category  Rating  \
7002   Adult Color by Number Book - Paint Mandala Pages   FAMILY     4.3   
5326                                      Al jazeera TV   FAMILY     4.6   
3874                       Rope Hero Return of a Legend     GAME     4.3   
10383                    Family Guy The Quest for Stuff     GAME     4.0   
6996               Art Pixel Coloring. Color by Number.   FAMILY     4.5   
...                                                 ...      ...     ...   
5843                           Mehmet Emin Ay İlahileri   FAMILY     NaN   
5846                                          YAKALA AY     GAME     NaN   
5855                      Ay Vamos - PJ. Balvin - Piano     GAME     NaN   
8580                                       DM Adventure     GAME     NaN   
9902                                         EU WHO War   FAMILY     NaN   

      Reviews                Size     Installs  Type  Price Content Rating  \
7002      997  Varies with device     100,000+  Free      0       Everyone   
5326     9952                3.7M     100,000+  Free      0           Teen   
3874     9951                 96M   1,000,000+  Free      0     Mature 17+   
10383  995002  Varies with device  10,000,000+  Free      0     Mature 17+   
6996      992                 49M     100,000+  Free      0           Teen   
...       ...                 ...          ...   ...    ...            ...   
5843        0                5.1M          10+  Free      0       Everyone   
5846        0                 14M           1+  Paid  $0.99       Everyone   
5855        0                 29M           5+  Free      0       Everyone   
8580        0                 11M          10+  Free      0       Everyone   
9902        0                1.2M          50+  Free      0       Everyone   

              Genres       Last Updated Current Ver   Android Ver  
7002   Entertainment      June 27, 2018         2.4    4.1 and up  
5326   Entertainment       July 5, 2016       1.0.0  4.0.3 and up  
3874          Arcade    August 26, 2016         1.0    2.3 and up  
10383      Adventure      July 25, 2018      1.73.0    4.1 and up  
6996   Entertainment     August 2, 2018   1.2.0.200    4.2 and up  
...              ...                ...         ...           ...  
5843   Entertainment      July 13, 2018         1.0    4.1 and up  
5846          Arcade       July 7, 2018         1.0    4.1 and up  
5855          Arcade       July 9, 2018         1.0    4.1 and up  
8580       Adventure      June 18, 2018       1.0.4    2.3 and up  
9902       Education  December 23, 2013         1.0    3.0 and up  

[2802 rows x 13 columns]

### Removal of paid apps

Recall that we only develop games that do not cost money to install, so including applications that do not fit this criteria would cause our analysis to be wrong. I will filter our datasets to only include free apps / apps with a price value of zero.

It is also important to note that the google dataset prefixes the prices with a dollar sign, so I remove them to match the apple dataset, and be able to convert the strings to a float.

In [21]:
google_df['Price'] = google_df['Price'].str.replace('$', '')
apple_df = apple_df.loc[apple_df['price'].astype(float) == 0.0]
google_df = google_df.loc[google_df['Price'].astype(float) == 0.0]

display(apple_df)
display(google_df)

id                                      track_name  size_bytes  \
2      529479190                                  Clash of Clans   116476928   
3      420009108                                      Temple Run    65921024   
7      553834731                                Candy Crush Saga   222846976   
9      343200656                                     Angry Birds   175966208   
10     512939461                                  Subway Surfers   156038144   
...          ...                                             ...         ...   
6595   883539642  ! OH Fantastic Free Kick + Kick Wall Challenge   162557952   
6599  1042978679                                       Tiny Bone   113205248   
6602  1043450394                    BrainTraining!!GASHAAAAAAN!!   132264960   
6585   967870967                                   Cyber Reversi    52166656   
6572   895700579                                      大富豪 Online   187623424   

     currency  price  rating_count_tot  rating_count_ver  user_rating  \
2         USD    0.0           2130805               579          4.5   
3         USD    0.0           1724546              3842          4.5   
7         USD    0.0            961794              2453          4.5   
9         USD    0.0            824451               107          4.5   
10        USD    0.0            706110                97          4.5   
...       ...    ...               ...               ...          ...   
6595      USD    0.0                 0                 0          0.0   
6599      USD    0.0                 0                 0          0.0   
6602      USD    0.0                 0                 0          0.0   
6585      USD    0.0                 0                 0          0.0   
6572      USD    0.0                 0                 0          0.0   

      user_rating_ver      ver cont_rating prime_genre  sup_devices.num  \
2                 4.5  9.24.12          9+       Games               38   
3                 4.0    1.6.2          9+       Games               40   
7                 4.5  1.101.0          4+       Games               43   
9                 3.0    7.4.0          4+       Games               38   
10                4.0   1.72.1          9+       Games               38   
...               ...      ...         ...         ...              ...   
6595              0.0      4.0          4+       Games               40   
6599              0.0    1.5.4          9+       Games               40   
6602              0.0    1.9.2          4+       Games               38   
6585              0.0    1.0.4          4+       Games               40   
6572              0.0  1.2.299          4+       Games               40   

      ipadSc_urls.num  lang.num  vpp_lic  
2                   5        18        1  
3                   5         1        1  
7                   5        24        1  
9                   0        10        1  
10                  5         1        1  
...               ...       ...      ...  
6595                5         2        1  
6599                5         9        1  
6602                2         1        1  
6585                5         4        1  
6572                5         1        1  

[1872 rows x 16 columns]

App Category  Rating  \
7002   Adult Color by Number Book - Paint Mandala Pages   FAMILY     4.3   
5326                                      Al jazeera TV   FAMILY     4.6   
3874                       Rope Hero Return of a Legend     GAME     4.3   
10383                    Family Guy The Quest for Stuff     GAME     4.0   
6996               Art Pixel Coloring. Color by Number.   FAMILY     4.5   
...                                                 ...      ...     ...   
5837             Ako ay may lobo Pinoy Kid Song Offline   FAMILY     NaN   
5843                           Mehmet Emin Ay İlahileri   FAMILY     NaN   
5855                      Ay Vamos - PJ. Balvin - Piano     GAME     NaN   
8580                                       DM Adventure     GAME     NaN   
9902                                         EU WHO War   FAMILY     NaN   

      Reviews                Size     Installs  Type Price Content Rating  \
7002      997  Varies with device     100,000+  Free     0       Everyone   
5326     9952                3.7M     100,000+  Free     0           Teen   
3874     9951                 96M   1,000,000+  Free     0     Mature 17+   
10383  995002  Varies with device  10,000,000+  Free     0     Mature 17+   
6996      992                 49M     100,000+  Free     0           Teen   
...       ...                 ...          ...   ...   ...            ...   
5837        0                9.7M           1+  Free     0     Mature 17+   
5843        0                5.1M          10+  Free     0       Everyone   
5855        0                 29M           5+  Free     0       Everyone   
8580        0                 11M          10+  Free     0       Everyone   
9902        0                1.2M          50+  Free     0       Everyone   

              Genres       Last Updated Current Ver   Android Ver  
7002   Entertainment      June 27, 2018         2.4    4.1 and up  
5326   Entertainment       July 5, 2016       1.0.0  4.0.3 and up  
3874          Arcade    August 26, 2016         1.0    2.3 and up  
10383      Adventure      July 25, 2018      1.73.0    4.1 and up  
6996   Entertainment     August 2, 2018   1.2.0.200    4.2 and up  
...              ...                ...         ...           ...  
5837   Entertainment      July 30, 2018         3.0    2.3 and up  
5843   Entertainment      July 13, 2018         1.0    4.1 and up  
5855          Arcade       July 9, 2018         1.0    4.1 and up  
8580       Adventure      June 18, 2018       1.0.4    2.3 and up  
9902       Education  December 23, 2013         1.0    3.0 and up  

[2538 rows x 13 columns]

We are left with 1872 apple applications and 2538 google games

# Analysis

Now that we have cleaned our data, we are ready to start the analysis.
Our companies strategy is to build a minimalistic Andriod verion of the app, while devloping it further if it has a good initial response. If it becomes profitable (from ad revenue) we develop an iOS version of the application.

Since a successful app for us means deployment on both operating systems, we need to find games that exist on both systems and are successful. We can to that by creating a new dataframe that consists of all games in one dataframe that are also present in the other dataframe. From there we will continue our analysis to identify that most successful games.

In [22]:
combined_df = google_df[google_df['App'].isin(apple_df['track_name'])].copy()
combined_df

App Category  Rating   Reviews                Size  \
1982      Zombie Catchers     GAME     4.7    990796                 75M   
1890    My Talking Angela     GAME     4.5   9883367                 99M   
1848   The Sims™ FreePlay     GAME     4.3    931595                 31M   
10010       Mobile Strike   FAMILY     3.9    903392  Varies with device   
1977    Hill Climb Racing     GAME     4.4   8923847                 63M   
...                   ...      ...     ...       ...                 ...   
3921          Rolling Sky     GAME     4.5   1117212                 50M   
7937       Shadow Fight 2     GAME     4.6  10981850                 88M   
1885                  Pou     GAME     4.3  10486018                 24M   
1764           Pokémon GO     GAME     4.1  10424925                 85M   
3975              Hay Day   FAMILY     4.5  10053186                 94M   

           Installs  Type Price Content Rating      Genres    Last Updated  \
1982    10,000,000+  Free     0       Everyone      Action    May 24, 2018   
1890   100,000,000+  Free     0       Everyone      Casual    July 3, 2018   
1848    10,000,000+  Free     0           Teen  Simulation   July 16, 2018   
10010   50,000,000+  Free     0   Everyone 10+    Strategy   June 14, 2018   
1977   100,000,000+  Free     0       Everyone      Racing    July 2, 2018   
...             ...   ...   ...            ...         ...             ...   
3921    50,000,000+  Free     0       Everyone       Board  August 3, 2018   
7937   100,000,000+  Free     0   Everyone 10+      Action    July 2, 2018   
1885   500,000,000+  Free     0       Everyone      Casual    May 25, 2018   
1764   100,000,000+  Free     0       Everyone   Adventure   July 23, 2018   
3975   100,000,000+  Free     0       Everyone      Casual   June 28, 2018   

              Current Ver   Android Ver  
1982               1.0.27    4.1 and up  
1890             3.7.2.51    4.1 and up  
1848               5.39.1  4.0.3 and up  
10010          3.30.4.207    4.1 and up  
1977               1.37.2    4.2 and up  
...                   ...           ...  
3921   Varies with device  4.0.3 and up  
7937               1.9.38    3.0 and up  
1885               1.4.77    4.0 and up  
1764              0.111.3    4.4 and up  
3975              1_39_93  4.0.3 and up  

[124 rows x 13 columns]

I used the columns from the google data because it has the granulated Genre column that will be useful in finding games of different types. We are almost left with a number of rows where we could make conclusions by just scanning the data, but I will break it down some more. The way I want to do this requires stripping the '+' from each value in the installs column.

In [23]:
combined_df.loc[:, 'Installs'] = combined_df['Installs'].str.replace('+', '')
combined_df.loc[:, 'Installs'] = combined_df['Installs'].str.replace(',', '').astype(int)
display(combined_df)

App Category  Rating   Reviews                Size  \
1982      Zombie Catchers     GAME     4.7    990796                 75M   
1890    My Talking Angela     GAME     4.5   9883367                 99M   
1848   The Sims™ FreePlay     GAME     4.3    931595                 31M   
10010       Mobile Strike   FAMILY     3.9    903392  Varies with device   
1977    Hill Climb Racing     GAME     4.4   8923847                 63M   
...                   ...      ...     ...       ...                 ...   
3921          Rolling Sky     GAME     4.5   1117212                 50M   
7937       Shadow Fight 2     GAME     4.6  10981850                 88M   
1885                  Pou     GAME     4.3  10486018                 24M   
1764           Pokémon GO     GAME     4.1  10424925                 85M   
3975              Hay Day   FAMILY     4.5  10053186                 94M   

        Installs  Type Price Content Rating      Genres    Last Updated  \
1982    10000000  Free     0       Everyone      Action    May 24, 2018   
1890   100000000  Free     0       Everyone      Casual    July 3, 2018   
1848    10000000  Free     0           Teen  Simulation   July 16, 2018   
10010   50000000  Free     0   Everyone 10+    Strategy   June 14, 2018   
1977   100000000  Free     0       Everyone      Racing    July 2, 2018   
...          ...   ...   ...            ...         ...             ...   
3921    50000000  Free     0       Everyone       Board  August 3, 2018   
7937   100000000  Free     0   Everyone 10+      Action    July 2, 2018   
1885   500000000  Free     0       Everyone      Casual    May 25, 2018   
1764   100000000  Free     0       Everyone   Adventure   July 23, 2018   
3975   100000000  Free     0       Everyone      Casual   June 28, 2018   

              Current Ver   Android Ver  
1982               1.0.27    4.1 and up  
1890             3.7.2.51    4.1 and up  
1848               5.39.1  4.0.3 and up  
10010          3.30.4.207    4.1 and up  
1977               1.37.2    4.2 and up  
...                   ...           ...  
3921   Varies with device  4.0.3 and up  
7937               1.9.38    3.0 and up  
1885               1.4.77    4.0 and up  
1764              0.111.3    4.4 and up  
3975              1_39_93  4.0.3 and up  

[124 rows x 13 columns]

It is just important to understand the the installs are saying there are AT LEAST that many, and not exactly that value. I want to get some basic statistics to try and see what some possible benchmarks could be.

In [24]:
combined_df['Installs'].describe()

count    1.240000e+02
mean     6.395645e+07
std      1.236198e+08
min      1.000000e+05
25%      1.000000e+07
50%      3.000000e+07
75%      1.000000e+08
max      1.000000e+09
Name: Installs, dtype: float64

While we could try and use a percentile to classify an app as 'successful', it is not my ideal approach for here. The apps are skewed and it would make more sence to just grab the top few games. I will keep the top 25% and that will leave me with a dataset that can be manually reviewed to make our final conclusions. Note that taking the top 25% is different than taking the top 25th percentile, as here we are guarenteed to have a quarter of the data.

In [25]:
combined_df = combined_df.loc[combined_df['Installs'] >= 100000000]
combined_df

App Category  Rating   Reviews                Size  \
1890        My Talking Angela     GAME     4.5   9883367                 99M   
1977        Hill Climb Racing     GAME     4.4   8923847                 63M   
3912      Asphalt 8: Airborne     GAME     4.5   8389714                 92M   
1920             Temple Run 2     GAME     4.3   8119154                 62M   
10186        Farm Heroes Saga   FAMILY     4.4   7615646                 71M   
1673                  PAC-MAN     GAME     4.2    685981                 37M   
1781             Trivia Crack     GAME     4.5   6427773                 95M   
1873    Candy Crush Soda Saga     GAME     4.4   6199095                 67M   
4188       Geometry Dash Lite     GAME     4.5   6181640                 58M   
1988   Hungry Shark Evolution     GAME     4.5   6074627                100M   
2004              Score! Hero     GAME     4.6   5419676                 96M   
1976            Traffic Racer     GAME     4.5   5387781                 53M   
1921               slither.io     GAME     4.4   5235294  Varies with device   
4037             Fruit Ninja®     GAME     4.3   5091448                 41M   
1914           Zombie Tsunami     GAME     4.4   4921451  Varies with device   
2007          Jetpack Joyride     GAME     4.4   4638163                 96M   
1879           Clash of Clans     GAME     4.6  44893888                 98M   
2206                   ROBLOX   FAMILY     4.5   4450890                 67M   
4059                Smash Hit     GAME     4.5   4147718                 79M   
4061               Temple Run     GAME     4.3   4000433                 42M   
1843            Angry Birds 2     GAME     4.6   3883589                 57M   
1947                  Agar.io     GAME     4.2   3816799                 32M   
1667               Sonic Dash     GAME     4.5   3778921                 75M   
1892            Cooking Fever     GAME     4.5   3198176                 82M   
1917           Subway Surfers     GAME     4.5  27725352                 76M   
1992      Hill Climb Racing 2     GAME     4.6   2750645  Varies with device   
1984          Angry Birds Rio     GAME     4.4   2610680                 46M   
1878             Clash Royale     GAME     4.6  23136735                 97M   
1869         Candy Crush Saga     GAME     4.4  22430188                 74M   
1908           My Talking Tom     GAME     4.5  14892469  Varies with device   
1981                Flow Free     GAME     4.3   1295625                 11M   
3897          Super Mario Run     GAME     3.7   1221896                 70M   
4890    Angry Birds Star Wars     GAME     4.3   1218055                 45M   
7937           Shadow Fight 2     GAME     4.6  10981850                 88M   
1885                      Pou     GAME     4.3  10486018                 24M   
1764               Pokémon GO     GAME     4.1  10424925                 85M   
3975                  Hay Day   FAMILY     4.5  10053186                 94M   

         Installs  Type Price Content Rating                        Genres  \
1890    100000000  Free     0       Everyone                        Casual   
1977    100000000  Free     0       Everyone                        Racing   
3912    100000000  Free     0           Teen                        Racing   
1920    500000000  Free     0       Everyone                        Action   
10186   100000000  Free     0       Everyone                        Casual   
1673    100000000  Free     0       Everyone                        Arcade   
1781    100000000  Free     0       Everyone                        Trivia   
1873    100000000  Free     0       Everyone                        Casual   
4188    100000000  Free     0       Everyone                        Arcade   
1988    100000000  Free     0           Teen                        Arcade   
2004    100000000  Free     0       Everyone                        Sports   
1976    100000000  Free     0       Everyone     

In [26]:
threshold = combined_df['Installs'].quantile(0.75)
combined_df = combined_df[combined_df['Installs'] >= threshold]
combined_df

App Category  Rating   Reviews                Size  \
1890        My Talking Angela     GAME     4.5   9883367                 99M   
1977        Hill Climb Racing     GAME     4.4   8923847                 63M   
3912      Asphalt 8: Airborne     GAME     4.5   8389714                 92M   
1920             Temple Run 2     GAME     4.3   8119154                 62M   
10186        Farm Heroes Saga   FAMILY     4.4   7615646                 71M   
1673                  PAC-MAN     GAME     4.2    685981                 37M   
1781             Trivia Crack     GAME     4.5   6427773                 95M   
1873    Candy Crush Soda Saga     GAME     4.4   6199095                 67M   
4188       Geometry Dash Lite     GAME     4.5   6181640                 58M   
1988   Hungry Shark Evolution     GAME     4.5   6074627                100M   
2004              Score! Hero     GAME     4.6   5419676                 96M   
1976            Traffic Racer     GAME     4.5   5387781                 53M   
1921               slither.io     GAME     4.4   5235294  Varies with device   
4037             Fruit Ninja®     GAME     4.3   5091448                 41M   
1914           Zombie Tsunami     GAME     4.4   4921451  Varies with device   
2007          Jetpack Joyride     GAME     4.4   4638163                 96M   
1879           Clash of Clans     GAME     4.6  44893888                 98M   
2206                   ROBLOX   FAMILY     4.5   4450890                 67M   
4059                Smash Hit     GAME     4.5   4147718                 79M   
4061               Temple Run     GAME     4.3   4000433                 42M   
1843            Angry Birds 2     GAME     4.6   3883589                 57M   
1947                  Agar.io     GAME     4.2   3816799                 32M   
1667               Sonic Dash     GAME     4.5   3778921                 75M   
1892            Cooking Fever     GAME     4.5   3198176                 82M   
1917           Subway Surfers     GAME     4.5  27725352                 76M   
1992      Hill Climb Racing 2     GAME     4.6   2750645  Varies with device   
1984          Angry Birds Rio     GAME     4.4   2610680                 46M   
1878             Clash Royale     GAME     4.6  23136735                 97M   
1869         Candy Crush Saga     GAME     4.4  22430188                 74M   
1908           My Talking Tom     GAME     4.5  14892469  Varies with device   
1981                Flow Free     GAME     4.3   1295625                 11M   
3897          Super Mario Run     GAME     3.7   1221896                 70M   
4890    Angry Birds Star Wars     GAME     4.3   1218055                 45M   
7937           Shadow Fight 2     GAME     4.6  10981850                 88M   
1885                      Pou     GAME     4.3  10486018                 24M   
1764               Pokémon GO     GAME     4.1  10424925                 85M   
3975                  Hay Day   FAMILY     4.5  10053186                 94M   

         Installs  Type Price Content Rating                        Genres  \
1890    100000000  Free     0       Everyone                        Casual   
1977    100000000  Free     0       Everyone                        Racing   
3912    100000000  Free     0           Teen                        Racing   
1920    500000000  Free     0       Everyone                        Action   
10186   100000000  Free     0       Everyone                        Casual   
1673    100000000  Free     0       Everyone                        Arcade   
1781    100000000  Free     0       Everyone                        Trivia   
1873    100000000  Free     0       Everyone                        Casual   
4188    100000000  Free     0       Everyone                        Arcade   
1988    100000000  Free     0           Teen                        Arcade   
2004    100000000  Free     0       Everyone                        Sports   
1976    100000000  Free     0       Everyone     

At first glance, there does not seem to be many top games classified as strategy, but I want to create a quick frequency table to better visualize the results.

In [27]:
combined_df['Genres'].value_counts()

Arcade                          13
Casual                           8
Action                           5
Racing                           4
Strategy                         2
Trivia                           1
Adventure                        1
Puzzle                           1
Adventure;Action & Adventure     1
Sports                           1
Name: Genres, dtype: int64

Only 2 of 37 (~5.4%) of top games are strategy games, and they are both by Supercell, a giant in the strategy game sector with games existing over a decade and making millions of dollars a month. Meanwhile, casual and arcade games make up the majority of games that we have determined as successful. 

I will take a quick look at some example games in each of those two genres before drawing conclusions.

In [28]:
combined_df.loc[(combined_df['Genres'] == 'Arcade') | (combined_df['Genres'] == 'Casual')]

App Category  Rating   Reviews                Size  \
1890        My Talking Angela     GAME     4.5   9883367                 99M   
10186        Farm Heroes Saga   FAMILY     4.4   7615646                 71M   
1673                  PAC-MAN     GAME     4.2    685981                 37M   
1873    Candy Crush Soda Saga     GAME     4.4   6199095                 67M   
4188       Geometry Dash Lite     GAME     4.5   6181640                 58M   
1988   Hungry Shark Evolution     GAME     4.5   6074627                100M   
4037             Fruit Ninja®     GAME     4.3   5091448                 41M   
1914           Zombie Tsunami     GAME     4.4   4921451  Varies with device   
2007          Jetpack Joyride     GAME     4.4   4638163                 96M   
4059                Smash Hit     GAME     4.5   4147718                 79M   
4061               Temple Run     GAME     4.3   4000433                 42M   
1843            Angry Birds 2     GAME     4.6   3883589                 57M   
1667               Sonic Dash     GAME     4.5   3778921                 75M   
1892            Cooking Fever     GAME     4.5   3198176                 82M   
1917           Subway Surfers     GAME     4.5  27725352                 76M   
1984          Angry Birds Rio     GAME     4.4   2610680                 46M   
1869         Candy Crush Saga     GAME     4.4  22430188                 74M   
1908           My Talking Tom     GAME     4.5  14892469  Varies with device   
4890    Angry Birds Star Wars     GAME     4.3   1218055                 45M   
1885                      Pou     GAME     4.3  10486018                 24M   
3975                  Hay Day   FAMILY     4.5  10053186                 94M   

         Installs  Type Price Content Rating  Genres       Last Updated  \
1890    100000000  Free     0       Everyone  Casual       July 3, 2018   
10186   100000000  Free     0       Everyone  Casual     August 7, 2018   
1673    100000000  Free     0       Everyone  Arcade       May 15, 2018   
1873    100000000  Free     0       Everyone  Casual      July 10, 2018   
4188    100000000  Free     0       Everyone  Arcade  December 22, 2017   
1988    100000000  Free     0           Teen  Arcade      July 25, 2018   
4037    100000000  Free     0       Everyone  Arcade      July 12, 2018   
1914    100000000  Free     0   Everyone 10+  Arcade      June 15, 2018   
2007    100000000  Free     0   Everyone 10+  Arcade      July 19, 2018   
4059    100000000  Free     0       Everyone  Arcade  November 26, 2015   
4061    100000000  Free     0       Everyone  Arcade     August 2, 2018   
1843    100000000  Free     0       Everyone  Casual      July 26, 2018   
1667    100000000  Free     0       Everyone  Arcade      July 26, 2018   
1892    100000000  Free     0       Everyone  Arcade      July 12, 2018   
1917   1000000000  Free     0   Everyone 10+  Arcade      July 12, 2018   
1984    100000000  Free     0       Everyone  Arcade       July 3, 2018   
1869    500000000  Free     0       Everyone  Casual       July 5, 2018   
1908    500000000  Free     0       Everyone  Casual      July 19, 2018   
4890    100000000  Free     0       Everyone  Arcade       May 25, 2018   
1885    500000000  Free     0       Everyone  Casual       May 25, 2018   
3975    100000000  Free     0       Everyone  Casual      June 28, 2018   

              Current Ver         Android Ver  
1890             3.7.2.51          4.1 and up  
10186               5.2.6          2.3 and up  
1673                6.6.3          4.1 and up  
1873              1.118.4          4.1 and up  
4188                  2.2          4.0 and up  
1988                6.0.0          4.1 and up  
4037         2.6.7.487220          4.1 and up  
1914   Varies with device  Varies with device  
2007              1.10.12          4.1 and up  
4059                1.4.0          2.3 and up  
4061                1.9.1          4.0 and up  
1843               2.21.1          4.1

There are plenty of familiar games on there, and some of the casual games could also be considered arcade games

# Conclusions

It seems like sticking strictly to strategy games is a high-risk, high-reward strategy. We need to balance risk & reward, so it would be beneficial to create games in additional genres. It appears like arcade games have a good chance of being successful, so I will reccomend we look into developing an arcade game.